### ISSUE TITLE
mkosi-mainline tests failing

### ISSUE THREAD

**Issue by solardiz** [lkrg-org/lkrg#283]
This has been happening for a while, @vt-alt:

```
Processing triggers for linux-image-unsigned-6.5.0-060500rc4daily20230731-generic (6.5.0-060500rc4daily20230731.202307302200) ...
/etc/kernel/postinst.d/dracut:
dracut: Generating /boot/initrd.img-6.5.0-060500rc4daily20230731-generic
/etc/kernel/postinst.d/zz-systemd-boot:
Installing kernel version 6.5.0-060500rc4daily20230731-generic in systemd-boot...
debconf: unable to initialize frontend: Dialog
debconf: (Dialog frontend will not work on a dumb terminal, an emacs shell buffer, or without a controlling terminal.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.36.0 /usr/local/share/perl/5.36.0 /usr/lib/x86_64-linux-gnu/perl5/5.36 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl-base /usr/lib/x86_64-linux-gnu/perl/5.36 /usr/share/perl/5.36 /usr/local/lib/site_perl) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7.)
debconf: falling back to frontend: Teletype
update-initramfs: deferring update (trigger activated)
dpkg: error: dpkg frontend lock was locked by another process with pid 5
Note: removing the lock file is always wrong, can damage the locked area
and the entire system. See <https://wiki.debian.org/Teams/Dpkg/FAQ#db-lock>.
run-parts: /etc/kernel/postinst.d/zz-systemd-boot exited with return code 2
dpkg: error processing package linux-image-unsigned-6.5.0-060500rc4daily20230731-generic (--install):
 installed linux-image-unsigned-6.5.0-060500rc4daily20230731-generic package post-installation script subprocess returned error exit status 1
Processing triggers for dracut (059-4) ...
dracut: Generating /boot/initrd.img-6.3.0-7-generic
dracut: Generating /boot/initrd.img-6.5.0-060500rc4daily20230731-generic
Errors were encountered while processing:
 linux-image-unsigned-6.5.0-060500rc4daily20230731-generic
Error: Workspace command /root/prepare final returned non-zero exit code 1.
```

**Comment by vt-alt** [comment]
> dpkg: error: dpkg frontend lock was locked by another process with pid 5

I tried to debug but without much success, this looks quite obscure (pid 5 is dpkg itself). Packages actually installed after faulty dpkg run but I fear we cannot just ignore exit status (1) of it because of other possible errors.

Idea 1: Maybe we can check if new kernel files are appeared then ignore `dpkg -i` errors.

**Comment by solardiz** [comment]
Fixed in #284.
